### PR TITLE
remote: Fix incorrect timeout issue and ssh cmd error

### PR DESCRIPTION
**Pull request by Yingshun**
The value of timeout was set to 'None' and verbose was directly
added to ssh command line in previous code, so fixup it in this PR.

Signed-off-by: Yingshun Cui <yicui@redhat.com>

### CODE REVIEW

**Comment by Yingshun** [comment]
**Test result:**
_Before fix:_
```
>>> runner_on_target = remote.RemoteRunner(host=serv_ip, username="root",password="redhat")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/var/tmp/libvirt-ci/runtest/avocado-vt/avocado-vt/virttest/remote.py", line 547, in __init__
    preferred_authenticaton)
  File "/var/tmp/libvirt-ci/runtest/avocado-vt/avocado-vt/virttest/remote.py", line 123, in wait_for_ssh_login_to_migrate
    internal_timeout, interface, verbose=verbose)
  File "/var/tmp/libvirt-ci/runtest/avocado-vt/avocado-vt/virttest/remote.py", line 87, in ssh_login_to_migrate
    handle_prompts(session, username, password, prompt, timeout)
  File "/root/.libvirt-ci-venv-ci-runtest-O7Ra8j/lib/python3.6/site-packages/aexpect/remote.py", line 262, in handle_prompts
    timeout=timeout, internal_timeout=0.5)
  File "/root/.libvirt-ci-venv-ci-runtest-O7Ra8j/lib/python3.6/site-packages/aexpect/client.py", line 934, in read_until_last_line_matches
    print_func)
  File "/root/.libvirt-ci-venv-ci-runtest-O7Ra8j/lib/python3.6/site-packages/aexpect/client.py", line 845, in read_until_output_matches
    end_time = time.time() + timeout
TypeError: unsupported operand type(s) for +: 'float' and 'str'

```

_After fix:_
```
>>> runner_on_target = remote.RemoteRunner(host=serv_ip, username="root",password="redhat")
>>> runner_on_target
<virttest.remote.RemoteRunner object at 0x7fde6a66a898>
>>> 
```

**Comment by Yingshun** [comment]
@dzhengfy  Would you please review this PR?

File: virttest/remote.py [hunk 1]
```diff
@@ -67,7 +67,7 @@ def ssh_login_to_migrate(client, host, port, username, password, prompt, linesep
                             linesep, log_filename, None, timeout,
                             interface, identity_file, status_test_command,
                             verbose, bind_ip)
-
+    verbose = "-vv" if verbose else ""
     cmd = ("ssh %s -o UserKnownHostsFile=%s "
            "-o StrictHostKeyChecking=no -p %s" %
            (verbose, user_known_hosts_file, port))
```
Comment (pevogam) [inline virttest/remote.py:73]: How about not dynamically changing the type of this variable and using something simpler:
```diff
    cmd = ("ssh %s -o UserKnownHostsFile=%s "
           "-o StrictHostKeyChecking=no -p %s" %
-           (verbose, user_known_hosts_file, port))
+           ("-vv" if verbose else "", user_known_hosts_file, port))
```

File: virttest/remote.py [hunk 2]
```diff
@@ -119,7 +119,7 @@ def wait_for_ssh_login_to_migrate(client, host, port, username, password, prompt
     while time.time() < end_time:
         try:
             return ssh_login_to_migrate(client, host, port, username, password, prompt,
-                                        linesep, log_filename, log_function,
```
Comment (pevogam) [inline virttest/remote.py:122]: Could we instead add the login function argument in `ssh_login_to_migrate`?
```diff
def ssh_login_to_migrate(client, host, port, username, password, prompt, linesep="\n",
-                         log_filename=None, timeout=10, interface=None, identity_file=None,
+                         log_filename=None, log_filename=None, timeout=10, interface=None, identity_file=None,
                         status_test_command="echo $?", verbose=False, bind_ip=None,
...
    if client != "ssh":
        return remote_login(client, host, port, username, password, prompt,
-                            linesep, log_filename, None, timeout,
+                            linesep, log_filename, log_function, timeout,
                            interface, identity_file, status_test_command,
```
? As you can guess from the name this function is new and temporary thus its API change should not matter too much and the change I propose is in the direction of this migration.

**Review by pevogam — COMMENTED** [review]
Thanks for the fix! I left a few comments that I want to hear your thoughts on.

File: virttest/remote.py (hunk 1 above)
Comment (Yingshun) [reply virttest/remote.py]: Updated.

**Comment by Yingshun** [comment]
It's introduced by 2e361394efde1468f9.

File: virttest/remote.py (hunk 2 above)
Comment (pevogam) [reply virttest/remote.py]: Sure, thanks for making the step forward!

File: virttest/remote.py [hunk 3]
```diff
@@ -513,7 +514,8 @@ class RemoteRunner(object):
     def __init__(self, client="ssh", host=None, port="22", username="root",
                  password=None, prompt=r"[\#\$]\s*$", linesep="\n",
                  log_filename=None, timeout=240, internal_timeout=10,
-                 session=None, preferred_authenticaton='password'):
+                 session=None, preferred_authenticaton='password',
+                 log_function=None):
```
Comment (pevogam) [inline virttest/remote.py:518]: This is actually another good step to make the functionality available for the `RemoteRunner` too, very good!

**Review by pevogam — APPROVED** [review]
LGTM now, thanks for the fixup!

**Review by dzhengfy — APPROVED** [review]
LGTM

**Pull request merged**